torch/csrc/jit/passes/batch_mm.cpp

```diff
@@ -432,7 +437,8 @@ void BatchMMSide(Block* block, AliasDb& alias_db) {
 
   std::unordered_set<Value*> considered_values;
   for (Node* node : block->nodes()) {
-    if (node->matches("aten::mm(Tensor self, Tensor mat2) -> Tensor")) {
+    if (node->matches("aten::mm(Tensor self, Tensor mat2) -> Tensor") &&
+        !alias_db.hasWriters(node)) {
       for (Value* input : node->inputs()) {
         if (/*bool not_inserted = */ !considered_values.emplace(input).second) {
           continue;
```

Comment:
Do we need to add a check in `removeTransposesAndGatherMatmuls` ?

Reply:
I guess not because this is just examining a tree we've already verified to not have writers